File: src/Entity/Evenement.php
<?php

namespace App\Entity;

use App\Repository\EvenementRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Vich\UploaderBundle\Mapping\Annotation as Vich;
use Symfony\Component\HttpFoundation\File\File;
use Symfony\Component\Serializer\Annotation\Groups;

/**
 * @ORM\Entity(repositoryClass=EvenementRepository::class)
 * @Vich\Uploadable
 */
class Evenement
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     * @Groups("ev")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Selectionnez le type  ! ")
     * @Groups("ev")
     */
    public $Type;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Le champ Nom evenement est vide ! ")
     * @Groups("ev")
     */
    private $NomEvenement;

    /**
     * @ORM\Column(type="date")
     * @Assert\GreaterThan("today", message ="La dade de début ne devrait pas être inférieure à la date du jour ! ")
     * @Groups("ev")
     */
    public $DateDebut;

    /**
     * @ORM\Column(type="date")
     * @Assert\Expression(
     *     "this.getDateDebut() < this.getDateFin()",
     *     message="La date fin ne doit pas  etre inférieure a la date de début "
     * )
     * @Groups("ev")
     */
    public $DateFin;


    /**
     * @ORM\Column(type="string", length=255)
     * *@Assert\NotBlank(message="Le champ LIEUX est vide ! ")
     * @Groups("ev")
     */
    private $Lieux;
    /**
     * @ORM\Column(type="string", length=255)
     * @var string
     * @Groups("ev")
     */
    private $image;

    /**
     * @Vich\UploadableField(mapping="product_images", fileNameProperty="image")
     * @var File
     * @Groups("ev")
     */
    private $imageFile;
    /**
     * @ORM\OneToMany(targetEntity=Offres::class, mappedBy="evenement",cascade={"remove"})
     */
    private $Offres;
    /**
     * @ORM\OneToMany(targetEntity=ReservationE::class, mappedBy="evenement")
     */
    private $reservationE;

    /**
     * @ORM\Column(type="string", length=100, nullable=true)
     */
    private $color;







    public function setImageFile($image = null)
    {
        $this->imageFile = $image;

        // VERY IMPORTANT:
        // It is required that at least one field changes if you are using Doctrine,
        // otherwise the event listeners won't be called and the file is lost
        if ($image) {
            // if 'updatedAt' is not defined in your entity, use another property
            $this->updatedAt = new \DateTime('now');
        }
    }

    public function getImageFile()
    {
        return $this->imageFile;
    }

    public function setImage($image)
    {
        $this->image = $image;
    }

    public function getImage()
    {
        return $this->image;
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getType(): ?string
    {
        return $this->Type;
    }

    public function setType(string $Type): self
    {
        $this->Type = $Type;

        return $this;
    }

    public function getNomEvenement(): ?string
    {
        return $this->NomEvenement;
    }

    public function setNomEvenement(string $NomEvenement): self
    {
        $this->NomEvenement = $NomEvenement;

        return $this;
    }

    public function getDateDebut(): ?\DateTimeInterface
    {
        return $this->DateDebut;
    }

    public function setDateDebut(\DateTimeInterface $DateDebut): self
    {
        $this->DateDebut = $DateDebut;

        return $this;
    }

    public function getDateFin(): ?\DateTimeInterface
    {
        return $this->DateFin;
    }

    public function setDateFin(\DateTimeInterface $DateFin): self
    {
        $this->DateFin = $DateFin;

        return $this;
    }

    public function getLieux(): ?string
    {
        return $this->Lieux;
    }

    public function setLieux(string $Lieux): self
    {
        $this->Lieux = $Lieux;

        return $this;
    }

    public function __toString()
    {
        return(string)$this->getType();
    }

    public function getColor(): ?string
    {
        return $this->color;
    }

    public function setColor(?string $color): self
    {
        $this->color = $color;

        return $this;
    }


}

File: src/entities/Evenement.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entities;

import java.util.Date;

/**
 *
 * @author Hp
 */
public class Evenement {
    
             int id ; 
            String Type ,NomEvenement   , image ,color , lieux  ; 
            Date DateDebut ,DateFin ; 
            public static String pathfile; 
            public static String filename="";

    public Evenement() {
    }

    public Evenement(int id, String Type, String NomEvenement, String image, String color, String lieux, Date DateDebut, Date DateFin) {
        this.id = id;
        this.Type = Type;
        this.NomEvenement = NomEvenement;
        this.image = image;
        this.color = color;
        this.lieux = lieux;
        this.DateDebut = DateDebut;
        this.DateFin = DateFin;
    }

    public Evenement(String Type, String NomEvenement, String image, String color, String lieux, Date DateDebut, Date DateFin) {
        this.Type = Type;
        this.NomEvenement = NomEvenement;
        this.image = image;
        this.color = color;
        this.lieux = lieux;
        this.DateDebut = DateDebut;
        this.DateFin = DateFin;
    }

   

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getType() {
        return Type;
    }

    public void setType(String Type) {
        this.Type = Type;
    }

    public String getNomEvenement() {
        return NomEvenement;
    }

    public void setNomEvenement(String NomEvenement) {
        this.NomEvenement = NomEvenement;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public String getLieux() {
        return lieux;
    }

    public void setLieux(String lieux) {
        this.lieux = lieux;
    }

    public Date getDateDebut() {
        return DateDebut;
    }

    public void setDateDebut(Date DateDebut) {
        this.DateDebut = DateDebut;
    }

    public Date getDateFin() {
        return DateFin;
    }

    public void setDateFin(Date DateFin) {
        this.DateFin = DateFin;
    }

    public static String getPathfile() {
        return pathfile;
    }

    public static void setPathfile(String pathfile) {
        Evenement.pathfile = pathfile;
    }

    public static String getFilename() {
        return filename;
    }

    public static void setFilename(String filename) {
        Evenement.filename = filename;
    }

    @Override
    public String toString() {
        return "Evenement{" + "id=" + id + ", Type=" + Type + ", NomEvenement=" + NomEvenement + ", image=" + image + ", color=" + color + ", lieux=" + lieux + ", DateDebut=" + DateDebut + ", DateFin=" + DateFin + '}';
    }

  
}
